lib/wheel.c

```diff
@@ -40,7 +40,6 @@ static int wheel_timer_thread_helper(struct thread *t)
 	void *data;
 
 	wheel = THREAD_ARG(t);
-	THREAD_OFF(wheel->timer);
```

Comment:
Why?

Reply:
I think it's not necessary and cost XCALLOC in 'thread_get'
It's already 'THREAD_UNUSED'

Reply:
I agree it's not really good practice to mess with the internals during a callback, but this isn't the real issue with the way the timerwheel uses the `lib/thread` apis? I don't see why we need `wheel_timer_thread` to get the timer callback and then throw a second 'thread' at `wheel_timer_thread_helper`. why not just have the timer callback be `helper` and eliminate the intermediate `thread_execute`?

Reply:
@lyq140 is correct here, the thread is off already.

Reply:
Would be worth updating then all places (I believe there are more with this pattern)?

Reply:
> I agree it's not really good practice to mess with the internals during a callback, but this isn't the real issue with the way the timerwheel uses the `lib/thread` apis? I don't see why we need `wheel_timer_thread` to get the timer callback and then throw a second 'thread' at `wheel_timer_thread_helper`. why not just have the timer callback be `helper` and eliminate the intermediate `thread_execute`?

Looks good.
Maybe you can submit another PR.

Reply:
> Would be worth updating then all places (I believe there are more with this pattern)?

Very likely.
Maybe you can go through the project.